### PR TITLE
Support IPEX framework

### PR DESCRIPTION
**Pull request by violetch24**
### what does this PR do？
This PR is used to support INC IPEX features and add QA example.

**examples/neural_compressor/question-answering/run_qa.py**
- Add an argument "backend" in _OptimizationArguments_ to specify which backend would be used.
- Change _calib_dataloader_  from _get_train_dataloader()_ to _get_eval_dataloader()_ when using INC IPEX backend, in order to keep the form of _example_inputs_ identical.

**optimum/intel/neural_compressor/optimization.py**
- Load _best_model.pt_ when using INC IPEX backend.

**optimum/intel/neural_compressor/quantization.py**
- Use _torch.jit.load_ to recover the int8 model quantized by INC IPEX backend, instead of _torch.load_.

**examples/neural_compressor/question-answering/README.md**
- Add ipex backend example.

**Example**:
The following example applies INC IPEX backend:
python run_qa.py \
    --model_name_or_path bert-large-uncased-whole-word-masking-finetuned-squad \
    --dataset_name squad \
    --apply_quantization \
    --quantization_approach static \
    --do_train \
    --do_eval \
    --verify_loading \
    --output_dir /tmp/squad_output \
    --backend ipex

**Environment for this PR:**
Python 3.8.13
transformers 4.23.1
datasets 2.6.1
intel-extension-for-pytorch 1.12.300
torch 1.12.1
**TIPS**: INC 1.14.2 would be published by Oct 28, 2022  and pip install would be available by then. Before that, please clone this repo https://github.com/intel/neural-compressor for this PR.

### CODE REVIEW

File: optimum/intel/neural_compressor/quantization.py [hunk 1]
```diff
@@ -1,379 +1,390 @@
...
-
-class IncQuantizedModelForXLNetLM(IncQuantizedModel):
-
-    TRANSFORMERS_AUTO_CLASS = XLNetLMHeadModel
-
-
-class IncQuantizedModelForVision2Seq(IncQuantizedModel):
-
-    TRANSFORMERS_AUTO_CLASS = AutoModelForVision2Seq
+#  Copyright 2022 The HuggingFace Team. All rights reserved.
```
Comment (hshen14) [inline optimum/intel/neural_compressor/quantization.py:1]: Did you follow the same coding style?

File: examples/neural_compressor/question-answering/README.md [hunk 2]
```diff
@@ -56,6 +56,21 @@ python run_qa.py \
...
+python run_qa.py \
+    --model_name_or_path bert-large-uncased-whole-word-masking-finetuned-squad \
+    --dataset_name squad \
+    --apply_quantization \
+    --quantization_approach static \
+    --do_train \
+    --do_eval \
+    --verify_loading \
+    --output_dir /tmp/squad_output \
+    --backend ipex
```
Comment (echarlaix) [inline examples/neural_compressor/question-answering/README.md:71]: Instead of adding an entire command line example, I think it would be great to have a sentence explaning that IPEX can be enabled by setting the backend argument accordingly

File: examples/neural_compressor/question-answering/run_qa.py [hunk 3]
```diff
@@ -802,15 +806,18 @@ def train_func(model):
...
         quant_approach = IncQuantizationMode(q8_config.get_config("quantization.approach"))
+        calib_dataloader = trainer.get_train_dataloader() if quant_approach == IncQuantizationMode.STATIC else None
         # torch FX used for post-training quantization and quantization aware training
         # dynamic quantization will be added when torch FX is more mature
         if quant_approach != IncQuantizationMode.DYNAMIC:
             if not training_args.do_train:
                 raise ValueError("do_train must be set to True for static and aware training quantization.")
+            if optim_args.backend == "ipex":
+                q8_config.set_config("model.framework", "pytorch_ipex")
+                calib_dataloader = trainer.get_eval_dataloader()
```
Comment (echarlaix) [inline examples/neural_compressor/question-answering/run_qa.py:817]: Why is the calibration dataset taken from the evaluation dataset when IPEX is enabled ?

File: optimum/intel/neural_compressor/quantization.py [hunk 4]
```diff
@@ -265,6 +265,12 @@ def from_pretrained(
         else:
             model_class._keys_to_ignore_on_load_missing.extend(missing_keys_to_ignore_on_load)
 
+        if os.path.exists(os.path.join(model_name_or_path, "best_model.pt")):
```
Comment (echarlaix) [inline optimum/intel/neural_compressor/quantization.py:268]: The `IncQuantizedModel` class should also allow the loading of models hosted on the Hugging Face Hub. Also our file name format is `pytorch_model.bin`

File: optimum/intel/neural_compressor/optimization.py [hunk 5]
```diff
@@ -139,6 +139,17 @@ def save_pretrained(self, save_directory: Optional[Union[str, os.PathLike]] = No
 
         os.makedirs(save_directory, exist_ok=True)
         self.config.save_pretrained(save_directory)
+        if isinstance(self._model.model, torch.jit._script.RecursiveScriptModule):
+            import json
+
+            try:
+                with open(os.path.join(save_directory, "best_configure.json"), "w") as f:
```
Comment (echarlaix) [inline optimum/intel/neural_compressor/optimization.py:146]: Is the `best_configure` json file is used when loading a quantized model ? If not I would remove saving it

File: examples/neural_compressor/question-answering/README.md (hunk 2 above)
Comment (violetch24) [reply examples/neural_compressor/question-answering/README.md:71]: Edited with a short explanation.

File: optimum/intel/neural_compressor/optimization.py (hunk 5 above)
Comment (violetch24) [reply optimum/intel/neural_compressor/optimization.py:146]: Since `best_configure.json` contains more information than the pt file, like data type of each ops, we recommend saving the json file.
